webplatformcompat/renderers.py

```diff
@@ -1,244 +1,242 @@
-            return getattr(resource.serializer, 'fields', None)
+                for error in value:
+                    fmt_error = self.dict_class((
+                        ('status', str(status_code)),
+                        ('detail', error),
+                        ('path', '/%s' % name),
+                    ))
+                    errors.append(fmt_error)
+        assert errors, data
+        return self.dict_class((('errors', errors),))
+
+    def convert_standard(self, data, fields_extra, main_resource, request):
+        """Convert a standard data item to RC1 format.
+
+        Keyword Arguments:
+        data - A dictionary-like object
+        fields_extra - Field metadata, keyed by field name
+        main_resource - The name of the resource key
+        request - Used to generate full URLs
+        """
+        attributes = self.dict_class()
+        link_ids = self.dict_class()
+        link_patterns = self.dict_class()
+        linked = self.dict_class()
+        meta = self.dict_class()
+
+        for name, value in data.items():
+            field_extra = fields_extra.get(name, {})
+            link = field_extra.get('link')
+            attr_name = field_extra.get('name', name)
+            if link:
+                link_pattern, link_id = self.convert_link(
+                    name, value, field_extra, main_resource, request)
+                if link_pattern:
+                    link_ids[attr_name] = link_id
+                    pattern_name = "%s.%s" % (main_resource, attr_name)
+                    link_patterns[pattern_name] = link_pattern
+                else:
+                    attributes[attr_name] = link_id
+            elif name == '_view_extra':
+                meta.update(value.pop('meta', {}))
+                for ve_key, ve_list in value.items():
+                    converted_list, resource = self.convert_list(
+                        ve_list, request)
+                    assert resource == ve_key, (
+                        "%s != %s" % (resource, ve_key))
+                    linked[ve_key] = converted_list[resource]
+                    link_patterns.update(converted_list.get('links', {}))
+            else:
+                attributes[attr_name] = value
+
+        converted = self.dict_class(((main_resource, attributes),))
+        if link_ids:
+            converted[main_resource]['links'] = link_ids
+        if link_patterns:
+            converted['links'] = link_patterns
+        if linked:
+            converted['linked'] = linked
+        if meta:
+            converted['meta'] = meta
+        return converted
+
+    def convert_paginated(self, data, request):
+        """Convert paginated data to JSON API RC1.
+
+        The pagination data is moved to the "meta" key, and the
+        paginated "results" are split into resources and link data.
+        """
+        keys = set(data.keys())
+        is_paginated = keys >= set(('count', 'next', 'previous', 'results'))
```

Comment:
On second though, this function only really applies to rendering, and the second check is redundant - ``convert_paginated`` is only called from ``render`` if this check passed. I'm adding a class instance method ``is_paginated`` to name the check (and so it can be used in the v2 API renderer) and dropping the assert.